internal/worker/httpserver/worker.go

```diff
@@ -312,7 +319,7 @@ func (w *Worker) newSimpleListener() (listener, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	w.logger.Infof(context.TODO(), "listening on %q", listener.Addr())
+	w.logger.Infof(context.Background(), "listening on %q", listener.Addr())
```

Comment:
Is a new background context really needed here? Should one not be passed in from above?

(Here & below)

Reply:
I wasn't going to change this one. If someone wants to do it, I'm fine for it to happen. I don't think it provides any or much value.